tools/frr-reload.py

```diff
@@ -424,7 +425,7 @@ def load_contexts(self):
                 continue
 
             # one line contexts
-            if new_ctx is True and any(line.startswith(keyword) for keyword in oneline_ctx_keywords):
+            if new_ctx is True and any(line.startswith(keyword) for keyword in oneline_ctx_keywords) and not (len(ctx_keys) > 0 and ctx_keys[0].startswith("mpls ldp") and line.startswith("router-id ")):
```

Comment:
nit: I just suggest rewriting this part without such a long if/for statement. It looks a bit ugly and confusing :) Maybe a proper comment what is done here would be awesome :+1:

Reply:
I can definitely add a comment! I am not sure how to break this statement though: the function is a huge sequence of `if...else if...else` to figure out what is the next step to be taken, and if I do not hit this condition I need to continue evaluating the others.